src/sharding/Shard.js

```diff
@@ -117,11 +140,25 @@ class Shard extends EventEmitter {
           execArgv: this.execArgv,
         })
         .on('message', this._handleMessage.bind(this))
-        .on('exit', this._exitListener);
+        .on('exit', this._exitListener)
+        .on('disconnect', this._handleDisconnect.bind(this, { parentProcessDisconnect: true }));
+
+      this.pid = this.process.pid;
+      this.ppid = this.process.ppid;
     } else if (this.manager.mode === 'worker') {
+      this.parentPort = require('worker_threads').parentPort;
       this.worker = new Worker(path.resolve(this.manager.file), { workerData: this.env })
         .on('message', this._handleMessage.bind(this))
         .on('exit', this._exitListener);
+
+      if (this.worker) {
+        setInterval(() => {
+          this.parentPort = require('worker_threads').parentPort;
+          if (!this.parentPort) {
+            this._handleDisconnect({ ParentThreadDisconnect: true });
```

Comment:
why capitalized p?

Reply:
Im assuming you're talking about `ParentThreadDisconnect`, tbf I don't know either

Reply:
corrected in [this commit](https://github.com/discordjs/discord.js/pull/5528/commits/cfb38e3a4725b99fce62afa0e6ddadcd2b373be5)